src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10030,11 +10040,19 @@ private bool ScanParenthesizedImplicitlyTypedLambda(Precedence precedence)
             }
 
             //  case 2:  ( x ) =>
-            if (IsTrueIdentifier(this.PeekToken(1))
-                && this.PeekToken(2).Kind == SyntaxKind.CloseParenToken
-                && this.PeekToken(3).Kind == SyntaxKind.EqualsGreaterThanToken)
+            if (IsTrueIdentifier(this.PeekToken(1)))
             {
-                return true;
+                if (this.PeekToken(2).Kind == SyntaxKind.CloseParenToken
+                && this.PeekToken(3).Kind == SyntaxKind.EqualsGreaterThanToken)
+                {
+                    return true;
+                }
+                if (this.PeekToken(2).Kind == SyntaxKind.ExclamationToken
+                && this.PeekToken(3).Kind == SyntaxKind.CloseParenToken
```

Comment:
>&& [](start = 16, length = 2)

Nit: Please align with the code inside the parens, not the beginning of the line.